### PR TITLE
Fix `update` operation for Tigris handler

### LINKED ISSUE #803: Support dot notation for update operators
Add dot notation support for:
* [x] `$inc`
* [x] `$set`
* [x] `$setOnInsert`
* [x] `$unset`

Add integration tests for each operator.

### PR DESCRIPTION
**Pull request by w84thesun**
# Description

This PR closes #803.

## Readiness checklist

* [x] I added tests for new functionality or bugfixes.
* [x] I ran `task all`, and it passed.
* [x] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by rumyantseva — APPROVED** [review]
The logic looks correct to me.

File: internal/handlers/tigris/msg_update.go [hunk 1]
```diff
@@ -195,24 +193,18 @@ func (h *Handler) MsgUpdate(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
...
 	h.L.Sugar().Debugf("Update: %s", u)
 
-	res, err := h.driver.UseDatabase(sp.db).Update(ctx, sp.collection, f, u)
+	_, err = h.driver.UseDatabase(sp.db).Replace(ctx, sp.collection, []driver.Document{u})
 	if err != nil {
 		return 0, lazyerrors.Error(err)
 	}
 
-	return int(res.ModifiedCount), nil
+	return 1, nil
```
Comment (noisersup) [inline internal/handlers/tigris/msg_update.go:209]: Isn't possible for Update to have more than 1 modified fields?

Comment (w84thesun) [reply internal/handlers/tigris/msg_update.go:209]: We return modified docs count, not fields. As we always modify only 1 document we must return 1.

Comment (noisersup) [reply internal/handlers/tigris/msg_update.go:209]: Understandable, LGTM then!

**Review by noisersup — APPROVED** [review]
LGTM (also sorry @rumyantseva) for re-requesting review by mistake...

**Review by rumyantseva — APPROVED** [review]
LGTM

**Review by AlekSi — COMMENTED** [review]
👍

**Pull request merged**